### DISCUSSION TITLE
Best way to avoid "response too long" error

### DISCUSSION THREAD

**Discussion by DanaMartens** [microsoft/vscode-discussions#2399] [Extension Development QnA] [answered]
I understand there are ways you provide to gracefully prune input tokens to stay within the max, but what is the best method to prevent the response from exceeding the token limit and causing a 'response too long' error? When I ask GitHub Copilot, it suggests the following (passing maxTokens) but it doesn't seem to be valid:

const chatResponse = await model.sendRequest(messages, { maxTokens: maxOutputTokens }, token);

**Comment by isidorn** [comment]
I thought prompt-tsx should help here. Maybe @roblourens has some advice.

**Reply by DanaMartens** [reply]
Any other suggestions?

**Comment by roblourens** [comment, marked as answer]
Each model has a certain response limit which can't be customized, and unfortunately there's no silver bullet for getting it to write a shorter response other than maybe just prompting it to write a shorter response :) We struggle with this too

**Comment by yosun** [comment]
this error doesn't seem to be dependent on model length - even when set to a smaller number it still occurs for OpenAI-compatible models on copilot
